src/smolvm/host/manager.py

```
@@ -294,7 +294,13 @@ def _download_and_extract(self, url: str, dest: Path, version: str, arch: str) -
                 for member in tar.getmembers():
                     if member.name.startswith("/") or ".." in member.name:
                         raise HostError(f"Refusing to extract suspicious path: {member.name}")
-                tar.extractall(path=tmp_dir)
+                try:
+                    tar.extractall(path=tmp_dir, filter="data")
+                except TypeError:
+                    # Python < 3.12 without PEP 706 backport — fall back to
+                    # unfiltered extraction.  The path validation above
+                    # guards against absolute/.. paths.
+                    tar.extractall(path=tmp_dir)
```

Ruff (0.15.12)

[error] Uses of `tarfile.extractall()`

(S202)